### DISCUSSION TITLE
How does the hook 'elementor/element/after_section_end' render values in the preview window?

### DISCUSSION THREAD

**Discussion by WebsiteDons** [elementor/elementor#18611] [Developers Community]
I injected fields to the elementor core columns using the hook, but the result is rendered only on the frontend and not in the preview window. **What hook is need to get the task done?**

## The coding to add the fields

```php
add_action('elementor/element/after_section_end', function($element,$section_id, $args)
{
  $name = $element->get_name();
  if( 'section_advanced' === $section_id )
  {
    if( 'column' === $name )
    {
      $element->start_controls_section('ctrl_section', [
        'label'=>'Set Column As Accordion',
        'tab'=> \Elementor\Controls_Manager::TAB_CONTENT
      ]);

        $element->add_control('col_acctitle',[
	  'type' => \Elementor\Controls_Manager::TEXT,
	  'label' => 'Title',
	  'label_block'=>true
	]);
        // more fields here

      $element->end_controls_section();
    }
  }
},10,3);
```
![2022-05-09 23_42_20](https://user-images.githubusercontent.com/42153624/167538508-1413a724-5a11-494e-bcd0-177c684fc9a2.jpg)

## Front end render

```php
add_action('elementor/frontend/column/before_render', function($element) 
{
  $v = (object)$element->get_settings_for_display();

  if( isset($v->col_accordion) && $v->col_accordion == 'yes' ) 
  {
    // other values processed here
    // title output
    echo '<h3 class="tab-handle" style="">'.$v->col_acctitle.'</h3>';
  }
});
```

## Front end display
The output is correct, but none of this shows in the elementor preview window
![2022-05-09 23_55_33](https://user-images.githubusercontent.com/42153624/167539720-796ff7a7-6ec0-4aa8-9395-a18bbabc0f43.jpg)

**Comment by WebsiteDons** [comment]
The following hook resolved the display.

```php
add_action('elementor/column/print_template', function($template, $element) 
{
  ob_start();
  ?>
    <#
      var maaxAcc = 'yes' === settings.col_accordion ? true : false;
      if( maaxAcc ) { #>
        <h3>the title</h3>
      <# } #>
  <?php
  $acc = ob_get_clean();

  return $acc . $template;
}
```

**Reply by m1ga** [reply]
Awesome, found your post because I had the same issue: how to render the new settings in the backend. Your code works fine :+1: 
Do you have an idea on how to add a class to the existing wrapper? I'm extending the `container` with `elementor/container/print_template` and I can add custom html tags like you did. But I would like to add custom class to the instead of adding html elements like you did. Any idea?
